docs/multi-cloud-deployment-guide.md

```diff
@@ -0,0 +1,489 @@
+# Multi-Cloud DocumentDB Deployment Guide
+
+This guide provides step-by-step instructions for setting up a multi-cloud
+deployment of DocumentDB using KubeFleet to manage clusters across
+clouds. This setup enables high availability and disaster recovery.
+We assume the use of an AKS cluster and an on-prem Kubernetes cluster
+that have network access to one another.
+
+## Table of Contents
+
+1. [Prerequisites](#prerequisites)
+2. [Architecture Overview](#architecture-overview)
+3. [Setting Up the Hub Cluster](#setting-up-the-hub-cluster)
+4. [Adding Clusters to Fleet](#adding-clusters-to-fleet)
+5. [Installing Operators and Dependencies](#installing-operators-and-dependencies)
+6. [Deploying DocumentDB Operator to Fleet](#deploying-documentdb-operator-to-fleet)
+7. [Setting Up Replication](#setting-up-replication)
+8. [Testing and Verification](#testing-and-verification)
+9. [Failover Procedures](#failover-procedures)
+
+## Prerequisites
+
+- Azure account
+- Azure CLI installed and configured with appropriate permissions
+- kubectl installed
+- helm installed
+- Git client
+- `kazure` alias like `alias kazure="kubectl --kubeconfig=/path/to/azure-kubeconfig"`
+- The same for `konprem`
+
+## Architecture Overview
+
+This multi-cloud deployment uses KubeFleet to manage DocumentDB instances across different cloud providers:
+
+- **Hub Cluster**: Central control plane for managing all member clusters
+- **Member Clusters**: Clusters in different cloud environments (Azure, On-prem)
+- **DocumentDB Operator**: Custom operator for DocumentDB deployments
+- **Fleet Networking**: Enables communication between clusters
+
+## Setting Up the Hub Cluster
+
+The hub cluster serves as the central controller for managing the member clusters, find setup instructions here:
+https://learn.microsoft.com/en-us/azure/kubernetes-fleet/overview
+
+## Adding Clusters to Fleet
+
+### Adding AKS cluster to the fleet
+
+Adding an AKS cluster to the fleet is very simple with the Azure portal: 
+https://learn.microsoft.com/en-us/azure/kubernetes-fleet/quickstart-create-fleet-and-members-portal
```

Comment:
Same. Should we render some commands here? If it's a lot, we can keep the link only.